Android/app/src/main/java/io/github/project_travel_mate/login/LoginActivity.java

```diff
@@ -403,6 +418,15 @@ public boolean validateEmail(String email) {
         }
     }
 
+    /**
+     * Checks if the device has an active network connection.
+     * @return Boolean return true if network connection is detected.
+     */
+    public boolean isNetworkConnected() {
```

Comment:
we already have a function in utils.java (https://github.com/project-travel-mate/Travel-Mate/blob/master/Android/app/src/main/java/utils/Utils.java#L67) to check for network connectivity. Please use that only.